[DPS-3091] Refactor of the Messaging > Rules and Conditions (#1056)
livechat/livechat-public-docs: src/pages/messaging/index.mdx

The next day, customer returns to ask about their other purchase. A new thread starts in the context of the chat, which has been **continuously open**. Each time the customer returns and starts a new conversation topic, they need to click **_start a chat_**. If we stuck to our naming convention, that would have to be **_start a thread_**, because this is what they actually do behind the scenes.

A **message** or **rich message** are not the only event types, though. There are also special events for specific actions, like: **file**, **filled form**, **system** and **custom** messages.

## Rules and conditions

Here are some general rules, which summarize the previous section and add new info.

1. When a new chat is started, **a new active thread** is created within this chat. When a customer returns and starts a new chat, new threads within a single chat are created on the server side.

2. There's always **only one active thread** in a certain chat. Only the **last thread** can be the **active** one. Events are always added to the active thread.

3. There can be time gaps between threads in a chat, but once a chat is started, it's **continuously open**.

> 4. If you have the [Messaging Mode](#messaging-mode) enabled on your licence, the messages can be sent and delivered even when the recipient (either customer or agent) is **offline**.

Reviewer: ```suggestion
4. If you have the [Messaging Mode](#messaging-mode) enabled for your licence, the messages can be sent and delivered even when the recipient (either customer or agent) is **offline**.
```

5. Since each chat consists of multiple different threads, **multiple agents** can participate in a single chat.

6. Every user can have **multiple concurrent chats**. Read how it applies to agents ([chatting as agent](#chatting-as-agent)) and customers ([chatting as customer](#chatting-as-customer)).

7. The algorithm that decides how chats are distributed between agents is called **routing**. It's documented in the [Routing](#chat-routing) section.

### Messaging Mode

> Once the Messaging Mode is enabled on your license, the chats become **continous**, which means once they're started, they remain open. Thanks to this feature, it's possible to receive and send messages to users when either an agent or the customer is offline. This way, the messaging becomes asynchronous and allows for more seamles communication. [Read more about the LiveChat Messaging Mode here](https://www.livechat.com/help/messaging-mode/).

Reviewer: ```suggestion
Once the Messaging Mode is enabled for your license, chats become **continuous**, which means once they're started, they remain open. Thanks to this mechanism, it's possible to receive and send messages to users when either an agent or the customer is offline. This way, messaging becomes asynchronous and allows for more seamless communication. [Read more about the LiveChat Messaging Mode here](https://www.livechat.com/help/messaging-mode/).
```

### Chatting as agent

The maximum number of concurrent chats an agent can participate in is **99**. It can be configured in the LiveChat Agent App by the license Owner or Admin.

### Chatting as customer

> One customer can only have one chat started, unless he chats with agents who use different licences. If you have the [Messaging Mode](https://www.livechat.com/help/messaging-mode/) enabled on your licence, the chats are continuous, and customers can always preview their chats' history. With **multiple concurrent chats** customers can sort their chats out thematically, making sure they know what each chat is about. If you don't use the Messaging Mode option on your licence, your licence's Owner or Admin can disable the chat history in the Agent App settings.

Reviewer: ```suggestion
One customer can only have one chat started unless he or she chats with agents who use different licences. If you have the [Messaging Mode](https://www.livechat.com/help/messaging-mode/) enabled for your licence, the chats are continuous, and customers can always preview their chat history. With **multiple concurrent chats**, customers can sort out their chats thematically, making sure they know what each chat is about. If you don't use the Messaging Mode option, your licence's Owner or Admin can disable the chat history in the Agent App settings.
```

Reviewer: 1. First sentence - I don't think "they" would go well with this sentence since we started it with "one customer".
2. I tried to avoid "on license" as it seems to be a literal translation from Polish.
3. I can't find it anywhere, but I'm pretty sure I've seen the recommendation to replace "Agent App", "LiveChat Agent App" with "LiveChat App".

Author: 1. Sure, I replaced this with a gender-neutral pronoun somewhat automatically :D 
2. 👍 
3. :O I believe that you might be right with this, maybe this is from our convo with Ola Kosicka about the naming conventions 🤔
